src/transformStateWithClones.js

```diff
@@ -7,7 +7,27 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const historyArray = [];
+  let changedObject = { ...state };
+
+  actions.forEach((action) => {
+    if (action.type === 'clear') {
```

Comment:
The switch statement should include a default case to handle any unexpected action types.